include/circt/Dialect/OM/Evaluator/Evaluator.h

```diff
@@ -64,6 +65,21 @@ struct AttributeValue : EvaluatorValue {
   Attribute attr;
 };
 
+struct IntegerValue : EvaluatorValue {
```

Comment:
Do we really need a new kind of EvaluatorValue? It seems like this could just be an AttributeValue. How is IntegerValue different from a normal AttributeValue, maybe I'm missing something, but I don't see it in this PR.

Reply:
Agreed, I think we don't need this.